spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/AppDeployerReleaseManager.java

```diff
@@ -167,19 +217,16 @@ public Release upgrade(Release existingRelease, Release replacingRelease, String
 		return status(release);
```

Comment:
Now that the upgradeStrategy would asynchronously wait for apps to be deployed, but since the `upgrade` call returns and the `status(release)` may have issues getting the status for the release that is getting asynchronously checked?

Reply:
I added kept the finder for AppDeploymentData to use the non-required version in status, in fact I ran into this race condition in testing.  It just returns the current release, with the idea the status method will be called again.

Reply:
Yeah, I see that. It makes sense :-)

Reply:
At least with plain install calling `status` updates platform status to the `Release`.